rcl/src/rcl/node_options.c

```diff
@@ -36,7 +36,7 @@ rcl_node_get_default_options()
     .arguments = rcl_get_zero_initialized_arguments(),
     .enable_rosout = true,
     .rosout_qos = rcl_qos_profile_rosout_default,
-    .enable_type_description_service = false,
+    .enable_type_description_service = true,
```

Comment:
As I've said in our weekly meetings, this is my biggest concern with this PR.  This means that for every new node a user constructs, we will now, by default, create the following services:

* `describe_parameters`
* `get_parameter_types`
* `get_parameters`
* `list_parameters`
* `set_parameters`
* `set_parameters_atomically`
* `get_type_description_service`

With each of those nominally taking 2 topics, that means that every node, by default, will now use 14 topics for services.  Add to that the default `/parameter_events`, and `/rosout` topics that are published/subscribed to, and we now have every node creating 16 topics before any user topics are included.  And given the fully-connected graph nature of DDS, that means a *lot* of discovery traffic at startup.

So the issue here is a system/integration issue, rather than anything particularly wrong with this PR.  There are a couple of ideas around this that we have been kicking around for a while, but that would definitely help mitigate this:

1.  All of those `*_parameter` services are excessive.  We honestly only need two: `get_parameters` and `set_parameters`.  `get_parameters`, with some appropriate configuration, could be used for `describe_parameters`, `get_parameter_types`, `get_parameters`, and `list_parameters`.  And `set_parameters` could be used for both `set_parameters` and `set_parameters_atomically`.  If we did this, then even with the addition of this PR we would be down to 8 topics per node.  This is my preferred solution to this issue, though it will probably be a bunch of work to fix this in the core.
2.  We have also had this idea of "lazy" services.  That is, the services would somehow advertise that they are available, but not actually go through the entire discovery process on startup.  They would only do full discovery when they want to be called.  I'm actually unsure of the details of how this one would work; @wjwwood might have more thoughts.

Reply:
While it's definitely not my preferred outcome - I would be open to an initial merge of this PR with `enable_type_description_service = false` by default.

Option 1 would be a good improvement on the current state, I agree it's gotten bloated.

Option 2 - "lazy" services would be great - since for _all_ of these services, I don't think anybody cares about them independently - you learn about the existence of a Node first, then you care about calling these node-specific services.

Reply:
After a lot of discussion, what we ended up deciding to do here is to merge this with the type description service disabled to start with.  Then we have a further plan to enable this by default, but adding in a parameter so it is easy to disable.  And then follow all of that up with lazy services.  So I'm going to resolve this particular conversation.